Zend/zend_operators.c

```diff
@@ -2972,7 +2972,7 @@ ZEND_API int ZEND_FASTCALL zend_binary_strncmp(const char *s1, size_t len1, cons
 	}
 	retval = memcmp(s1, s2, MIN(length, MIN(len1, len2)));
 	if (!retval) {
-		return (int)(MIN(length, len1) - MIN(length, len2));
+		return ZEND_THREEWAY_COMPARE(MIN(length, len1), MIN(length, len2));
```

Comment:
Since the macro uses the inputs twice, might it make sense to store the result of `MIN` into a tmp variable?

Reply:
Checked and this indeed does get appropriately optimized.